### PR TITLE
Dynamically load new goal data.

### PR DESCRIPTION
**Pull request by Zarabadoo**
Load goal data via ajax.

### CODE REVIEW

File: js/acquia_lift.reports.js [hunk 1]
```diff
@@ -1000,158 +1000,77 @@ Rickshaw.Graph.TableLegend = Rickshaw.Class.create(Rickshaw.Graph.Legend, {
...
-
-              for (var i = 0; i < input.length; i++) {
-                $wrapper
-                  .append(input[i]);
+            $data = $statistics.find('table[data-lift-statistics]'),
+            campaign = $data.attr('data-acquia-lift-campaign'),
+            $goalSelect = $('.acquia-lift-report-section-options .form-item-goal select'),
+            $metricSelect = $('.acquia-lift-report-section-options .form-item-metric select'),
+            metric = $metricSelect.val(),
+            $metricOptions = $metricSelect.children('option'),
```
Comment (eshtadc) [inline js/acquia_lift.reports.js]: $metricOptions doesn't look like it's used.

File: js/acquia_lift.reports.js [hunk 2]
```diff
@@ -1000,158 +1000,77 @@ Rickshaw.Graph.TableLegend = Rickshaw.Class.create(Rickshaw.Graph.Legend, {
 (function ($, Drupal) {
   "use strict";
 
-  /**
```
Comment (eshtadc) [inline js/acquia_lift.reports.js:1003]: Love how much code gets cleared out :-)

File: js/acquia_lift.reports.js [hunk 3]
```diff
@@ -1000,158 +1000,77 @@ Rickshaw.Graph.TableLegend = Rickshaw.Class.create(Rickshaw.Graph.Legend, {
...
-                $wrapper
-                  .append(input[i]);
+            $data = $statistics.find('table[data-lift-statistics]'),
+            campaign = $data.attr('data-acquia-lift-campaign'),
+            $goalSelect = $('.acquia-lift-report-section-options .form-item-goal select'),
+            $metricSelect = $('.acquia-lift-report-section-options .form-item-metric select'),
+            metric = $metricSelect.val(),
+            $metricOptions = $metricSelect.children('option'),
+            // Find the tqble column that matches the metric value.
+            metricColumn = function() {
```
Comment (eshtadc) [inline js/acquia_lift.reports.js:1014]: Can you simplify this by calling index() on the th that meets the selection criteria of the correct conversion metric?

File: js/acquia_lift.reports.js [hunk 4]
```diff
@@ -1000,158 +1000,77 @@ Rickshaw.Graph.TableLegend = Rickshaw.Class.create(Rickshaw.Graph.Legend, {
...
+        // Load a new report if the goal is changed.
+        $goalSelect.change(function() {
+          // Construct the GET path.
+          var path = Drupal.settings.basePath + 'acquia_lift/reports/conversion',
+              position = 0,
+              args = {
+                campaign: campaign,
+                goal: $(this).val()
+              };
+          for (var arg in args) {
```
Comment (eshtadc) [inline js/acquia_lift.reports.js]: Check out http://api.jquery.com/jQuery.param/ for simplification of this loop code.  As long as we're below 1.4.1 then it's fair game as far as I'm concerned ;-)
